src/components/Receive/AddressView.js

```diff
@@ -1,338 +1,98 @@
-        walletMeta.walletImplementationId,
-        walletMeta.networkId,
-        getCardanoByronConfig().PROTOCOL_MAGIC,
-        address,
-        addressingInfo,
-        hwDeviceInfo,
-        useUSB,
-      )
-    } catch (e) {
-      if (e instanceof LocalizableError) {
-        await showErrorDialog(errorMessages.generalLocalizableError, intl, {
-          message: intl.formatMessage(
-            {
-              id: e.id,
-              defaultMessage: e.defaultMessage,
-            },
-            e.values,
-          ),
-        })
-      } else {
-        Logger.error(e)
-        await showErrorDialog(errorMessages.hwConnectionError, intl, {
-          message: String(e.message),
-        })
-      }
-    } finally {
-      closeDetails()
-    }
-  })
-}
+  address: string,
+  onPressDetails: () => mixed,
+|}
 
-const MESSAGE_TIMEOUT = 1000
-const ADDRESS_DIALOG_STEPS = {
-  CLOSED: 'CLOSED',
-  ADDRESS_DETAILS: 'ADDRESS_DETAILS',
-  CHOOSE_TRANSPORT: 'CHOOSE_TRANSPORT',
-  ADDRESS_VERIFY: 'ADDRESS_VERIFY',
-  LEDGER_CONNECT: 'LEDGER_CONNECT',
+const AddressView = ({isUsed, index, address, onPressDetails}: Props) => {
+  return (
+    <Row>
+      <Address>
+        <Text secondary={isUsed} small bold>{`/${index}`}</Text>
+        <Text secondary={isUsed} small numberOfLines={1} ellipsizeMode="middle" monospace style={styles.text}>
+          {address}
+        </Text>
+      </Address>
+
+      <Actions>
+        <CopyButton text={address} />
+        <VerifyButton onPress={() => onPressDetails()} />
+      </Actions>
+    </Row>
+  )
 }
-type AddressDialogSteps = $Values<typeof ADDRESS_DIALOG_STEPS>
 
-type Props = {|
-  index: number,
-  addressInfo: AddressDTOCardano,
-  isUsed: boolean,
-  walletMeta: $Diff<WalletMeta, {id: string}>,
-  openDetails: () => void,
-  closeDetails: () => void,
-  onVerifyAddress: () => void,
-  addressDialogStep: AddressDialogSteps,
-  onToggleAddrVerifyDialog: () => void,
-  openTransportSwitch: () => void,
-  onChooseTransport: (Object, boolean) => void,
-  openAddressVerify: () => void,
-  useUSB: boolean,
-  isWaiting: boolean,
-  onConnectBLE: (DeviceId) => void,
-  onConnectUSB: (DeviceObj) => void,
-  intl: IntlShape,
-|}
+export default AddressView
 
-const AddressView = ({
-  index,
-  addressInfo,
-  isUsed,
-  walletMeta,
-  openDetails,
-  closeDetails,
-  onVerifyAddress,
-  addressDialogStep,
-  onToggleAddrVerifyDialog,
-  onChooseTransport,
-  useUSB,
-  isWaiting,
-  onConnectUSB,
-  onConnectBLE,
-  intl,
-}: Props) => {
+const Row = (props: ViewProps) => <View {...props} style={styles.container} />
+const Address = (props) => <View {...props} style={styles.addressContainer} />
+const Actions = (props) => <View {...props} style={styles.actionContainer} />
+const CopyButton = ({text}: {text: string}) => {
+  const MESSAGE_TIMEOUT = 1000
   const [isCopying, setIsCopying] = useState<boolean>(false)
 
-  const address = addressInfo.address
   useEffect(() => {
```

Comment:
Perhaps replace it with [this](https://github.com/Emurgo/yoroi-mobile/blob/develop/src/utils/useCopy.js)